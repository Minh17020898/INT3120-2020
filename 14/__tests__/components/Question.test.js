import React from 'react';
import renderer from 'react-test-renderer';
import Question from '../../components/Question';

describe('test question', () => {
  const question = renderer.create(
  <Question 
    question='test'
    answer1='true'
    answer2='false'
    answer3='false'
    result='one'
  />);
  const questionInstance = question.getInstance();
  const questionJSON = question.toJSON();

  // test 1
  it('select the second anwser', () => {
    questionInstance.twoPressed();
    expect(questionInstance.state).toEqual({
      one:false,
      two:true,
      three:false,
      correct:false,
      checked:false,
      key:''
    });
  });

  // test 2
  it('check result', () => {
    questionInstance.check('one');
    expect(questionInstance.state).toEqual({
      one:false,
      two:true,
      three:false,
      correct:false,
      checked:true,
      key:'one'
    });
  });

  // test 3
  it('renders correctly', () => {
    expect(questionJSON).toMatchSnapshot();
  });
})