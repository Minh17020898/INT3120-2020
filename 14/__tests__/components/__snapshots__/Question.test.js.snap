// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test question renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "black",
      "borderRadius": 8,
      "borderWidth": 1,
      "marginLeft": 16,
      "marginRight": 16,
      "paddingBottom": 8,
      "paddingLeft": 8,
      "paddingTop": 8,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
      }
    }
  >
    test
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "margin": 5,
          "marginLeft": 10,
          "marginRight": 10,
          "opacity": 1,
          "padding": 10,
        }
      }
      testID="checkbox"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text />
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "marginTop": 18,
          },
          null,
        ]
      }
    >
      true
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "margin": 5,
          "marginLeft": 10,
          "marginRight": 10,
          "opacity": 1,
          "padding": 10,
        }
      }
      testID="checkbox"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text />
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "marginTop": 18,
          },
          null,
        ]
      }
    >
      false
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "margin": 5,
          "marginLeft": 10,
          "marginRight": 10,
          "opacity": 1,
          "padding": 10,
        }
      }
      testID="checkbox"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text />
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "marginTop": 18,
          },
          null,
        ]
      }
    >
      false
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "width": "100%",
      }
    }
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          }
        }
      >
        <Text />
      </View>
    </View>
    <View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
</View>
`;
